dashboard/test/controllers/programming_expressions_controller_test.rb

```diff
@@ -25,4 +25,21 @@ class ProgrammingExpressionsControllerTest < ActionController::TestCase
     end
     assert @response.body.include? "Valid programming environment is required"
   end
+
+  test 'can update programming expression from params' do
+    sign_in @levelbuilder
+
+    programming_expression = create :programming_expression
+    post :update, params: {
+      id: programming_expression.id,
+      key: programming_expression.key,
+      name: 'new name',
+      shortDescription: 'short description of code'
+    }
+    assert_response :ok
+    programming_expression.reload
+
+    assert_equal 'new name', programming_expression.name
+    assert_equal 'short description of code', programming_expression.short_description
+  end
```

Comment:
This seems a bit light for unit tests covering the two new server endpoints. I would suggest at a minimum adding `test_user_gets_response_for` test cases to make sure only levelbuilders can hit these endpoints, and a simple test case like [this one](https://github.com/code-dot-org/code-dot-org/blob/5acb2a08747e1718dd6576612447af0a45c5496e/dashboard/test/controllers/lessons_controller_test.rb#L422-L446) verifying that the summarized data makes it into the edit page DOM.

I know you are trying to keep this PR small, but I think separating new functionality from unit tests if anything means there is more for us all to keep in our minds rather than less. If you've already got these in a future PR, then that's fine to keep them there and avoid rearranging.

Reply:
I agree with what you're implicitly saying about unit tests being included with the feature. I'll add some more here. Not trying to make this PR smaller by skimping on tests, just the fact that there is very little on this new page :) 

It also turns out there is a bug in the authorization -- but that's also true of a couple other models. I'll fix (and test) for programming expressions in this PR then follow-up with the other couple I noticed

Reply:
> I agree with what you're implicitly saying about unit tests being included with the feature. I'll add some more here. 

Thank you Bethany, that's a much clearer way of saying it. New tests look great!